core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java

```diff
@@ -403,6 +403,18 @@ private void applyUpdateInode(UpdateInodeEntry entry) {
     if (entry.hasPinned() && inode.isFile()) {
       if (entry.getPinned()) {
         MutableInodeFile file = inode.asFile();
+        List<String> mediaList = ServerConfiguration.getList(
+            PropertyKey.MASTER_TIERED_STORE_GLOBAL_MEDIUMTYPES, ",");
+        if (entry.getMediumTypeList().isEmpty()) {
+          // if user does not specify a pinned media list, any location is OK
+          file.setMediumTypes(new HashSet<>());
+        } else {
+          for (String medium : entry.getMediumTypeList()) {
+            if (mediaList.contains(medium)) {
+              file.getMediumTypes().add(medium);
+            }
```

Comment:
Should we throw an exception if the medium type is invalid?

Reply:
unfortunately we can't safely throw exceptions in this context, while we are in `applyUpdateInode`.   Maybe we can move the check to outside of this section.

Reply:
I'm not too concerned, we will just quietly do a best effort, it might be worth doing a preemptive check on the client side and warn if the global types don't match (but not fail).